econml/_ortho_learner.py

```diff
@@ -0,0 +1,584 @@
+        assert (sample_weight is None) or (sample_weight.shape[0] == Y.shape[0]), "Dimension mis-match!"
+        assert (sample_var is None) or (sample_var.shape[0] == Y.shape[0]), "Dimension mis-match!"
+        self._d_x = X.shape[1:] if X is not None else None
+
+    def _check_fitted_dims(self, X):
+        if X is None:
+            assert self._d_x is None, "X was not None when fitting, so can't be none for effect"
+        else:
+            assert self._d_x == X.shape[1:], "Dimension mis-match of X with fitted X"
+
+    def _subinds_check_none(self, var, inds):
+        return var[inds] if var is not None else None
+
+    def _filter_none_kwargs(self, **kwargs):
+        non_none_kwargs = {}
+        for key, value in kwargs.items():
+            if value is not None:
+                non_none_kwargs[key] = value
+        return non_none_kwargs
+
+    @BaseCateEstimator._wrap_fit
+    def fit(self, Y, T, X=None, W=None, Z=None, sample_weight=None, sample_var=None, *, inference=None):
+        """
+        Estimate the counterfactual model from data, i.e. estimates function :math:`\\theta(\\cdot)`.
+
+        Parameters
+        ----------
+        Y: (n, d_y) matrix or vector of length n
+            Outcomes for each sample
+        T: (n, d_t) matrix or vector of length n
+            Treatments for each sample
+        X: optional (n, d_x) matrix or None (Default=None)
+            Features for each sample
+        W: optional (n, d_w) matrix or None (Default=None)
+            Controls for each sample
+        Z: optional (n, d_z) matrix or None (Default=None)
+            Instruments for each sample
+        sample_weight: optional (n,) vector or None (Default=None)
+            Weights for each samples
+        sample_var: optional (n,) vector or None (Default=None)
+            Sample variance for each sample
+        inference: string, `Inference` instance, or None
+            Method for performing inference.  This estimator supports 'bootstrap'
+            (or an instance of `BootstrapInference`).
+
+        Returns
+        -------
+        self : _OrthoLearner instance
+        """
+        self._check_input_dims(Y, T, X, W, Z, sample_weight, sample_var)
+        nuisances, fitted_inds = self._fit_nuisances(Y, T, X, W, Z, sample_weight=sample_weight)
+        self._fit_final(self._subinds_check_none(Y, fitted_inds),
+                        self._subinds_check_none(T, fitted_inds),
+                        X=self._subinds_check_none(X, fitted_inds),
+                        W=self._subinds_check_none(W, fitted_inds),
+                        Z=self._subinds_check_none(Z, fitted_inds),
+                        nuisances=tuple([self._subinds_check_none(nuis, fitted_inds) for nuis in nuisances]),
+                        sample_weight=self._subinds_check_none(sample_weight, fitted_inds),
+                        sample_var=self._subinds_check_none(sample_var, fitted_inds))
```

Comment:
I'd avoid the repetition:

    [Y, T, X, W, Z, sample_weight, sample_var, *nuisances] = [self._subinds_check_none(arr, fitted_inds) 
                                                              for arr in [Y, T, X, W, Z, sample_weight, sample_var, *nuisances]]

Reply:
The nuisances is problematic here, as we want to apply the subinds function on each entry in the nuisance tuple. Don't think the above would work. Seems fine to keep it explicity for now.